### PR TITLE
feat: Format deploy code

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

File: script/DeploySpark.s.sol [hunk 1]
```diff
@@ -70,52 +70,96 @@ contract DeploySpark is Script {
...
-        registry = new PoolAddressesProviderRegistry(deployer);
+
+        // 1. Deploy and configure registry and addresses provider
+
+        registry              = new PoolAddressesProviderRegistry(deployer);
         poolAddressesProvider = new PoolAddressesProvider(config.readString(".marketId"), deployer);
+
         poolAddressesProvider.setACLAdmin(deployer);
+
+        // 2. Deploy and data provider and pool configurator
```
Comment (hexonaut) [inline script/DeploySpark.s.sol:98]: "Deploy and configure data provider..." missing word. Also, there is no mention of the initialization.

File: script/DeploySpark.s.sol [hunk 2]
```diff
@@ -70,52 +70,96 @@ contract DeploySpark is Script {
...
+
         protocolDataProvider = new AaveProtocolDataProvider(poolAddressesProvider);
-        PoolConfigurator _poolConfigurator = new PoolConfigurator();
-        _poolConfigurator.initialize(poolAddressesProvider);
-        Pool _pool = new Pool(poolAddressesProvider);
-        _pool.initialize(poolAddressesProvider);
+        poolConfigurator     = new PoolConfigurator();
```
Comment (hexonaut) [inline script/DeploySpark.s.sol:101]: I would do the same as you did with `pool` / `poolImpl` and save this to `poolConfiguratorImpl`,

File: script/DeploySpark.s.sol [hunk 3]
```diff
@@ -70,52 +70,96 @@ contract DeploySpark is Script {
...
+
         collectorImpl.initialize(address(0));
+
         (treasury, treasuryImpl) = createCollector(admin);
 
-        InitializableAdminUpgradeabilityProxy incentivesProxy = new InitializableAdminUpgradeabilityProxy();
-        incentives = RewardsController(address(incentivesProxy));
-        emissionManager = new EmissionManager(deployer);
-        RewardsController rewardsController = new RewardsController(address(emissionManager));
+        // 11. Deploy initialize and configure rewards contracts
```
Comment (hexonaut) [inline script/DeploySpark.s.sol:156]: No mention of setting the flash loan fee to zero.

File: script/DeploySpark.s.sol [hunk 4]
```diff
@@ -125,69 +169,77 @@ contract DeploySpark is Script {
...
-        ScriptTools.exportContract(instanceId, "uiPoolDataProvider", address(uiPoolDataProvider));
+        ScriptTools.exportContract(instanceId, "incentives",      address(incentives));
+        ScriptTools.exportContract(instanceId, "incentivesImpl",  address(rewardsController));
+        ScriptTools.exportContract(instanceId, "pool",            address(pool));
+
+        ScriptTools.exportContract(instanceId, "poolAddressesProvider",         address(poolAddressesProvider));
+        ScriptTools.exportContract(instanceId, "poolAddressesProviderRegistry", address(registry));
+
+        ScriptTools.exportContract(instanceId, "poolConfigurator",        address(poolConfigurator));
+        ScriptTools.exportContract(instanceId, "poolConfiguratorImpl",    address(poolConfigurator));
```
Comment (hexonaut) [inline script/DeploySpark.s.sol:224]: This should reference the pool configurator implementation - not the instance.

File: script/DeploySpark.s.sol [hunk 5]
```diff
@@ -125,69 +169,77 @@ contract DeploySpark is Script {
...
+        ScriptTools.exportContract(instanceId, "poolImpl",                address(pool));
```
Comment (hexonaut) [inline script/DeploySpark.s.sol:225]: This is also incorrectly referencing the pool instance instead of the implementation.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Reviewed the deploy script and there is some fixes needed. Doing test base next.

File: test/SparkDeployBase.t.sol [hunk 6]
```diff
@@ -77,75 +77,86 @@ abstract contract SparkDeployBaseTest is Test {
...
 
     function test_spark_deploy_aclManager() public {
         // NOTE: Also verify that no other address than the admin address has any role (verify with events)
         assertEq(address(aclManager.ADDRESSES_PROVIDER()), address(poolAddressesProvider));
-        assertTrue(aclManager.hasRole(aclManager.DEFAULT_ADMIN_ROLE(), admin));
-        assertTrue(!aclManager.hasRole(aclManager.DEFAULT_ADMIN_ROLE(), deployer));
-        assertEq(aclManager.getRoleAdmin(aclManager.POOL_ADMIN_ROLE()), aclManager.DEFAULT_ADMIN_ROLE());
-        assertTrue(aclManager.hasRole(aclManager.POOL_ADMIN_ROLE(), admin));
+
+        bytes32 defaultAdmin = aclManager.DEFAULT_ADMIN_ROLE();
```
Comment (hexonaut) [inline test/SparkDeployBase.t.sol:143]: Similarly probably makes sense to cache the `aclManager.POOL_ADMIN_ROLE()` as well.

File: test/SparkDeployBase.t.sol [hunk 7]
```diff
@@ -77,75 +77,86 @@ abstract contract SparkDeployBaseTest is Test {
...
+        assertEq(aclManager.getRoleAdmin(aclManager.EMERGENCY_ADMIN_ROLE()), defaultAdmin);
+
+        assertTrue( aclManager.hasRole(defaultAdmin, admin));
+        assertTrue(!aclManager.hasRole(defaultAdmin, deployer));
+
+        assertTrue( aclManager.hasRole(aclManager.POOL_ADMIN_ROLE(), admin));
         assertTrue(!aclManager.hasRole(aclManager.POOL_ADMIN_ROLE(), deployer));
-        assertEq(aclManager.getRoleAdmin(aclManager.EMERGENCY_ADMIN_ROLE()), aclManager.DEFAULT_ADMIN_ROLE());
+
         if (block.chainid == 1) assertTrue(aclManager.hasRole(aclManager.EMERGENCY_ADMIN_ROLE(), admin));     // FIXME missing on GOERLI
```
Comment (hexonaut) [inline test/SparkDeployBase.t.sol:154]: This conditional should be removed. There shouldn't be discrepencies between chains.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Some more changes in the test.

File: script/DeploySpark.s.sol (hunk 1 above)
Comment (lucas-manuel) [reply script/DeploySpark.s.sol:98]: Updated

File: script/DeploySpark.s.sol (hunk 2 above)
Comment (lucas-manuel) [reply script/DeploySpark.s.sol:101]: Updated

File: test/SparkDeployBase.t.sol (hunk 7 above)
Comment (lucas-manuel) [reply test/SparkDeployBase.t.sol]: test

**Pull request merged**